pkg/roachprod/vm/vm.go

```diff
@@ -41,13 +41,35 @@ const (
 	// TagArch is the CPU architecture tag const.
 	TagArch = "arch"
 
-	ArchARM64 = CPUArch("arm64")
-	ArchAMD64 = CPUArch("amd64")
-	ArchFIPS  = CPUArch("fips")
+	ArchARM64   = CPUArch("arm64")
+	ArchAMD64   = CPUArch("amd64")
+	ArchFIPS    = CPUArch("fips")
+	ArchUnknown = CPUArch("unknown")
 )
 
 type CPUArch string
 
+// ParseArch parses a string into a CPUArch using a simple, non-exhaustive heuristic.
```

Comment:
Would be nice to include some examples of what kinds of inputs this is known to work well: e.g., GCE tools, binary detection tool, etc.